### PR TITLE
Travis: install lua and luarocks with hererocks

### PR DESCRIPTION
**Pull request by zer0main**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,14 +4,13 @@ dist: trusty # modern Wt
...
-    - LUA=lua5.2
-    - LUA=lua5.3
-    - LUA=luajit     # latest stable version (2.0.4)
-    - LUA=luajit2.0  # current head of 2.0 branch
-    - LUA=luajit2.1  # current head of 2.1 branch
+    - LUA="lua5.1"
+    - LUA="lua5.2"
+    - LUA="lua5.3"
+    - LUA="luajit2.0"
+    - LUA="luajit2.1"
```
Comment (starius) [inline .travis.yml]: s/lua5.1/lua 5.1/

File: .travis.yml [hunk 2]
```diff
@@ -4,14 +4,13 @@ dist: trusty # modern Wt
 
 env:
   global:
-    - LUAROCKS=2.2.2
+    - LUAROCKS=2.3.0
```
Comment (starius) [inline .travis.yml]: Unrelated change, please discard.

File: .travis.yml [hunk 3]
```diff
@@ -20,8 +19,10 @@ branches:
 before_install:
   - sudo apt-get update
   - sudo apt-get install --yes libwt-dev libwthttp-dev
+  - sudo pip install hererocks
+  - hererocks here --$LUA --luarocks $LUAROCKS
+  - eval $(./here/bin/luarocks path --bin)
```
Comment (starius) [inline .travis.yml:25]: Move these commands after `sudo pip install cpp-coveralls` to preserve order.

File: .travis/setup_lua.sh [hunk 4]
```diff
@@ -1,124 +0,0 @@
...
-
-if [ "$LUAJIT" == "yes" ]; then
-  rm -rf $LUAJIT_BASE;
-elif [ "$LUA" == "lua5.1" ]; then
-  rm -rf lua-5.1.5;
-elif [ "$LUA" == "lua5.2" ]; then
-  rm -rf lua-5.2.4;
-elif [ "$LUA" == "lua5.3" ]; then
-  rm -rf lua-5.3.1;
-fi
```
Comment (starius) [inline .travis/setup_lua.sh:124]: I want this old shell file back.
</joke>

**Comment by zer0main** [comment]
We have some strange problems with busted. [FIXED]

**Comment by starius** [comment]
We should fix `.travis/valgrind_test.sh` at first. It seems to be incompatible with new version of busted.

**Comment by starius** [comment]
I am not happy with this workaround with `spec/*`.
We should identify a root reason of the failure. Could you reproduce it locally in clean environment?

**Comment by zer0main** [comment]
I think this failure has no relation to luawt. luarocks installs the latest version of busted `rc12`. I have `rc11` locally and no errors occurs.
You can check these commits:
https://github.com/Olivine-Labs/busted/commit/52cc01016dbb6a43510613663b0adf29291453d4
https://github.com/Olivine-Labs/busted/commit/0796e7fd1cc942da1a14152b3c1e945fd8c442d9
